### DISCUSSION TITLE
Companion app not registering - I've tried everything recommended

### DISCUSSION THREAD

**Discussion by VikkaVita** [aclap-dev/video-downloadhelper#43] [Help and Q&A] [answered]
### Your question

Hi there.
At the end of last year I had this problem with VDH companion app not registering. It turned on a yes/no switch not being set properly. Thank you to everyone who helped, especially Paul. You save my and my friends  studies.
I have the same problem again, but the switch is now properly set.
I've searched the board and have no found any solution. 
PC:
i5, running fully up to date Ubuntu 22.04.3 LTS
Browser: Firefox, snap. 123.0b2 (64-bit) Up to date.
This is what I have done, without success:
Removed VDH app, /opt folder etc. Searched for any left via name and removed.
Rebooted PC
Updated PC. All up to date
Reinstalled app from firefox extensions repository
Ran:
flatpak permissions webextensions
webextensions net.downloadhelper.coapp snap.firefox yes         0x00
Checked VDH Settings:
Checking companion app returned: No such native application net.downloadhelper.coapp. [This was to be expected. The coapp has not been installed.]
Used these commands from Paul: [@ https://github.com/aclap-dev/video-downloadhelper/discussions/37]
wget https://github.com/aclap-dev/vdhcoapp/releases/latest/download/vdhcoapp-linux-x86_64.deb
sudo dpkg -i ./vdhcoapp-linux-x86_64.deb
flatpak permission-set webextensions net.downloadhelper.coapp snap.firefox yes
Verified yes:

Then I went to VDH settings and:
Checking companion app returned: No such native application net.downloadhelper.coapp.
Closed F/F and re-opened it.
Then I went to VDH setting and:
Checking companion app returned: No such native application net.downloadhelper.coapp.
Then I ran:
/opt/vdhcoapp/vdhcoapp install
Output:
Installing…
Flatpak is installed. Making the coapp available from browser sandboxes:
Linked coapp within org.mozilla.firefox.
Linked coapp within com.brave.Browser.
Linked coapp within com.google.Chrome.
Linked coapp within com.google.ChromeDev.
Linked coapp within org.chromium.Chromium.
Linked coapp within com.github.Eloston.UngoogledChromium.
Linked coapp within com.microsoft.Edge.
Linked coapp within com.microsoft.EdgeDev.
Writing /home/fablemaker/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
Writing /home/fablemaker/.var/app/org.mozilla.firefox/.mozilla/native-messaging-hosts/net.downloadhelper.coapp.json
Writing /home/fablemaker/.config/google-chrome/NativeMessagingHosts/net.downloadhelper.coapp.json
Writing /home/fablemaker/.config/chromium/NativeMessagingHosts/net.downloadhelper.coapp.json
Writing /home/fablemaker/.config/BraveSoftware/Brave-Browser/NativeMessagingHosts/net.downloadhelper.coapp.json
Writing /home/fablemaker/.config/opera/NativeMessagingHosts/net.downloadhelper.coapp.json
VdhCoApp : VdhCoApp is ready to be used

Checking companion app returned: No such native application net.downloadhelper.coapp.
Ran:
/opt/vdhcoapp/vdhcoapp --info

Output

{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.15",
  "binary": "/opt/vdhcoapp/vdhcoapp",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "linux",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "/home/fablemaker",
  "converterBinary": "/opt/vdhcoapp/ffmpeg",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "63b936e"
}

Checking companion app returned: No such native application net.downloadhelper.coapp.

Closed F/F and rebooted

Checking companion app returned: No such native application net.downloadhelper.coapp.

**Comment by paulrouget** [comment, marked as answer]
> Updated PC. All up to date

Even Snap Core?

**Reply by VikkaVita** [reply]
Thanks Paul, for your reply.
Yes. They are. But....
I powered down the machine, rather than rebooted. I had to go out for a while and when I returned, I powered up the machine and the app is now recognised. I've never seen that before.
Thanks for your reply. 
So, the query is closed. Thanks also for your detailed instructions.
